packages/main/src/plugin/kubernetes/contexts-manager.spec.ts

```diff
@@ -2884,3 +2884,50 @@ describe('isContextChanged', () => {
     expect(routeInformer).toHaveBeenCalledOnce();
   });
 });
+
+describe('refreshContextState', () => {
+  test('should reset informers', async () => {
+    vi.mocked(makeInformer).mockImplementation(fakeMakeInformer);
+    const client = new TestContextsManager(apiSender);
+    const kubeConfig = new kubeclient.KubeConfig();
+    const config = {
+      clusters: [
+        {
+          name: 'cluster1',
+          server: 'server1',
+        },
+      ],
+      users: [
+        {
+          name: 'user1',
+        },
+      ],
+      contexts: [
+        {
+          name: `context1`,
+          cluster: 'cluster1',
+          user: 'user1',
+        },
+      ],
+      currentContext: 'context1',
+    };
+
+    kubeConfig.loadFromOptions(config);
+    await client.update(kubeConfig);
+    const expectedMap = new Map<string, CheckingState>();
+    expectedMap.set('context1', {
+      state: 'checking',
+    } as CheckingState);
+    vi.advanceTimersToNextTimer();
+    vi.advanceTimersToNextTimer();
+    expect(apiSenderSendMock).toHaveBeenCalledWith('kubernetes-contexts-checking-state-update', expectedMap);
+    apiSenderSendMock.mockClear();
+    vi.advanceTimersByTime(9000);
+    expect(apiSenderSendMock).not.toHaveBeenCalled();
+    vi.advanceTimersByTime(1100);
+    expect(apiSenderSendMock).toHaveBeenCalledWith('kubernetes-contexts-checking-state-update', expectedMap);
+    apiSenderSendMock.mockClear();
+    await client.refreshContextState('context1');
+    expect(apiSenderSendMock).toHaveBeenCalledWith('kubernetes-contexts-checking-state-update', expectedMap);
```

Comment:
looks like it's not checking when the context we ask to refresh is not there

![image](https://github.com/user-attachments/assets/42b519bf-6f1b-4f96-a3d5-8ba86370ed74)

but then, I'm wondering, should it be a no-op or an error to ask to refresh something that is not existing ?

Reply:
Yes, that's a good point, I'm not sure either. I have chosen no-op here, as IMO there is no chance that the refresh is asked if the context does not exist (but it will depend on the frontend implementation).

And if really the context has been deleted in the meantime, it seems to me that it would not be an error, because the user just doesn't care anymore about it.